polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -73,17 +73,19 @@ pub struct FetchedCollation {
 	/// Candidate hash.
 	pub candidate_hash: CandidateHash,
 	/// Id of the collator the collation was fetched from.
-	pub collator_id: CollatorId,
+	/// This will be missing when we create the struct from
+	/// a v2 candidate receipt. instance
+	pub collator_id: Option<CollatorId>,
```

Comment:
`FetchedCollation` is used as a key in a HashMap (`fetched_candidates`). I think we still need some collator identifier here to differentiate between two collators advertising the same collation.

Since this is a fetched collation, we should have access to the collator id which is declared in `CollatorProtocolMessage::Declare`

Reply:
Took me a while to remember why I did this. The gist of it is that for v2 descriptors, you cannot have the same candidate with different candidate hash, because there is no collator id and singature fields. This mapping is currently needed only for reputation changes and sending a Seconded message back to the collator that provided the collation. In slot based collator and even look ahead we don't make use of the seconded message AFAIK, and the reputation system just doesn't make much sense given that currently there is no way to really punish them.

Reply:
@ordian @eskimor any opinions on this change ?

Reply:
NVM, we fetch one collation at a time, so we'd not worry about having a collator id field at all here.